### ISSUE TITLE
Integrating ability to have multiple windows into Qline 

### ISSUE THREAD

**Issue by antonszilasi** [ladybug-tools/spider#59]
@theo-armour 
Unfortunately I am struggling to get the the multiple window functionality working with the qline code,

I managed to do it easily with r12 please see[ here](https://github.com/antonszilasi/spiderAnton/blob/be698208f6cd5f295e61f862ef89bcc08907117a/build-well/r12/build-well-threejs.html#L423-L472)

The closest I have got is the code [here](https://github.com/antonszilasi/spiderAnton/blob/master/speed/geometry.js#L795-L825
)
Which you can see here: https://antonszilasi.github.io/spiderAnton/speed/speed-page.html
As you can see although I do have multiple windows they all appear onto of each other.

How can I space these windows out over their parent surface, can you please provide an example.

**Comment by theo-armour** [comment]
@antonszilasi @bwelle 

I can't begin to count the number of silly things I have created in this project. This started from the get-go when I spent a day or so creating a [NURBS chart]( http://www.ladybug.tools/spider/sandbox/nurbs-chart/r2/nurbs-chart-random-numbers.html ) cookbook example when what you needed was a [scatter plot]( http://www.ladybug.tools/spider/sandbox/scatter-well/r1/index.html ) example. So let's help me not make a mistake this time.

I understand your struggles with the various gbXML building generators. I have them too.  As far as I know there are zero examples of open source code available for this type of thing. We start from scratch. We make mistakes. We fix or we end game / go back to start.

So in order to move fast and effectively, let's help me build it right.

What are the inputs?

* Everything in feet/inches/%?
* HTML input type=range or input type = number?
    * Decision required for each numerical input
    * If input type=number indicate if you want error checking for entering letter or negative numbers etc
* A window to wall ratio of percent with 1% min and 99%?
    * If you want 0% and 100% then these are likely to be special cases where there is an external wall with no opening or a wall that is entirely opening. Such cases are likely to have impact on the exported gbXML let alone the 3D display
* The number of openings per elevation
    * Zero openings and large numbers of openings are likely to create special cases
* Overhang depth per opening with min max and step
* Fin depth per opening with min max and step
* Dimension and location inputs and constraints
    * Choose one of the following:
        * Length, height, center point, azimuth and tilt?
       * Three x,y,z points?
    * Vertical walls only?
    * Orthogonal/ tied to X-axis or Y-axis walls only?
* Any data particulars - relating perhaps to gbXML or whatever - that should be associated any of the openings
 
Once we've agree the above. And added the bits I forgot about, we can list and verify the outcomes we desire,

I would like to build a cookbook code sample as a response so a description of the problem we are trying to solve and how a solution might bring benefit would be totally cool.

**Comment by antonszilasi** [comment]
@theo-armour 
@bwelle 

Actually after a fresh look at your qline code this morning - I managed to figure it out with the code:
https://github.com/antonszilasi/spiderAnton/commit/4b83e57aeef46ac6b8db53307e441425b2e523c1

Please see: https://antonszilasi.github.io/spiderAnton/speed/speed-page.html

I realize now that you were interpolating to place the windows, I will now go through and add the overhangs and fins as well as try and change the material propertys to make it more pretty.

At the moment what I think would be really helpful is if you could document your qline code as its pretty hard to follow in certain places. 

Thanks

**Comment by theo-armour** [comment]
@antonszilasi 

> At the moment what I think would be really helpful is if you could document your qline code as its pretty hard to follow in certain places.

Welcome to linear algebra.

I do my best to create scripts that require minimal documentation because the functions and variables are so exquisitely named and flow so logically from one to the next that if you know plain vanilla JavaScript, HTML and CSS and you are familiar with the basic principles of linear algebra and you can follow [Mr.doob's examples]( https://threejs.org/examples/ ) you should be able to read my code. If you see any place in my code where names or flow could be improved I want to hear. 

In any case, writing documentation for code that is already out of data does not appear to be could use of anybody's time.

In any case, bravo for finding a solution. Keep at it!

**Comment by bwelle** [comment]
Sounds good. I will put this all together tonight

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by theo-armour** [comment]
@antonszilasi @bwelle 

[wall with multiple openings overhangs fins]( http://www.ladybug.tools/spider/cookbook/multiple-openings/wall-with-multiple-openings-overhangs-fins-r1.html )

**Comment by bwelle** [comment]
@theo-armour @antonszilasi 

Gentlemen. It feels far but we are close. Most of my confusion is due why there is confusion. The gbXML export is actually very straight forward. I have done everything I can to be very specific in what needs to happen throughout a wide range of issues and on Confluence. Apparently, I was not as clear as I had hoped, and I know jumping back and forth between things is challenging. Let me try to take a step back and simply as much as possible for the final time, hopefully this will all make sense.

Please the cookbook [SPEED Geometry Requirements](https://github.com/ladybug-tools/spider/tree/master/cookbook/SPEED%20Geometry%20Requirement) and [SPEED gbXML Requirements](https://github.com/ladybug-tools/spider/tree/master/cookbook/SPEED%20gbXML%20Requirements)

**Comment by theo-armour** [comment]
And I thought this issues was about

> Integrating ability to have multiple windows into Qline
